File: Go/binarysearch.go
// Binary Search in Go

package main

import (
	"fmt"
)

func main() {

	var arr = []int{2, 3, 4, 10, 40}
	var x = 10
	fmt.Println(binarySearch(arr, x))

}

func binarySearch(arr []int, x int) int {
	low := 0
	high := len(arr) - 1 
	for low < high{
		median := (low + high) / 2

		if arr[median] < x{
			low = median + 1
		}else{
			high = median - 1
		}
	if arr[low] == x{
		return low
	}
	}
	
	return -1 // return index or -1 if element is not found
}
	


// Output:

// 3

File: Javascript/binarysearch.js
// Binary Search Algorithm in Javascript

function binarySearch(arr, x) {
    let start=0, end=arr.length-1;
         
    
    while (start<=end){
 
        let mid=Math.floor((start + end)/2);
  
       
        if (arr[mid]===x) return true;
 
        else if (arr[mid] < x)
             start = mid + 1;
        else
             end = mid - 1;
    }
  
    return false;
}

var arr = [2, 3, 4, 10, 40];

const x = 10;

var result = binarySearch(arr, x);

console.log(result);

// Output:
// true